**Linux** [Build 8706](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=77162&buildTypeId=WFPR) outcome was **FAILURE** using a merge of fd1ed48485953cb664d34011972dba7c0c84bf34
Summary: Tests failed: 1 (1 new), passed: 3282, ignored: 363 Build time: 00:51:32
### Failed tests
```
org.jboss.as.test.integration.jpa.jarfile.JpaJarFileTestCase: 	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:83)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
Caused by: java.lang.Exception: {"WFLYCTL0080: Failed services" => {
    "jboss.module.service.\"deployment.jpajarfile.ear.my-ejb-module.jar\".main" => "org.jboss.msc.service.StartException in service jboss.module.service.\"deployment.jpajarfile.ear.my-ejb-module.jar\".main: WFLYSRV0179: Failed to load module: deployment.jpajarfile.ear.my-ejb-module.jar:main
    Caused by: org.jboss.modules.ModuleLoadException: Error loading module from /opt/buildAgent/work/a31d203e70e89f90/dist/target/wildfly-10.0.0.CR5-SNAPSHOT/modules/system/layers/base/org/hibernate/4.3/module.xml
    Caused by: org.jboss.modules.xml.XmlPullParserException: Unknown attribute \"services\" from namespace \"\" (position: START_TAG seen ...ct name=\"${org.wildfly:jipijapa-hibernate4-3}\" services=\"import\"/>... @28:82) ",
    "jboss.module.service.\"deployment.jpajarfile.ear\".main" => "org.jboss.msc.service.StartException in service jboss.module.service.\"deployment.jpajarfile.ear\".main: WFLYSRV0179: Failed to load module: deployment.jpajarfile.ear:main
    Caused by: org.jboss.modules.ModuleNotFoundException: org.hibernate:4.3"
}}
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getActionResult(ServerDeploymentPlanResultFuture.java:134)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getResultFromNode(ServerDeploymentPlanResultFuture.java:123)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:85)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:42)
	at org.jboss.as.controller.client.helpers.standalone.ServerDeploymentHelper.deploy(ServerDeploymentHelper.java:55)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:77)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
------- Stdout: -------
 [0m19:08:24,947 INFO  [stdout] (pool-393-thread-1) Hibernate: insert into Employee (address, name, id) values (?, ?, ?)
 [0m [0m19:08:24,948 INFO  [stdout] (pool-393-thread-1) Hibernate: select employee0_.id as id1_0_0_, employee0_.address as address2_0_0_, employee0_.name as name3_0_0_ from Employee employee0_ where employee0_.id=?
 [0m [0m19:08:24,952 INFO  [org.jboss.weld.deployer] (MSC service thread 1-4) WFLYWELD0010: Stopping weld service for deployment jpa_sessionfactory.jar
 [0m [0m19:08:24,954 INFO  [org.jboss.as.jpa] (ServerService Thread Pool -- 28) WFLYJPA0011: Stopping Persistence Unit (phase 2 of 2) Service 'jpa_sessionfactory.jar#mypc'
 [0m [0m19:08:24,954 INFO  [org.hibernate.tool.hbm2ddl.SchemaExport] (ServerService Thread Pool -- 28) HHH000227: Running hbm2ddl schema export
 [0m [0m19:08:24,954 INFO  [stdout] (ServerService Thread Pool -- 28) Hibernate: drop table Employee if exists
 [0m [0m19:08:24,954 INFO  [org.hibernate.tool.hbm2ddl.SchemaExport] (ServerService Thread Pool -- 28) HHH000230: Schema export complete
 [0m [33m19:08:24,955 WARN  [org.hibernate.internal.SessionFactoryRegistry] (ServerService Thread Pool -- 28) HHH000374: Could not unbind factory from JNDI: org.hibernate.engine.jndi.JndiException: Error performing unbind [modelSessionFactory]
	at org.hibernate.engine.jndi.internal.JndiServiceImpl.unbind(JndiServiceImpl.java:213)
	at org.hibernate.internal.SessionFactoryRegistry.removeSessionFactory(SessionFactoryRegistry.java:124)
	at org.hibernate.internal.SessionFactoryImpl.close(SessionFactoryImpl.java:1077)
	at org.hibernate.jpa.internal.EntityManagerFactoryImpl.close(EntityManagerFactoryImpl.java:339)
	at org.jboss.as.jpa.service.PersistenceUnitServiceImpl$2$1.run(PersistenceUnitServiceImpl.java:220)
	at org.jboss.as.jpa.service.PersistenceUnitServiceImpl$2$1.run(PersistenceUnitServiceImpl.java:206)
	at org.wildfly.security.manager.WildFlySecurityManager.doChecked(WildFlySecurityManager.java:667)
	at org.jboss.as.jpa.service.PersistenceUnitServiceImpl$2.run(PersistenceUnitServiceImpl.java:239)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jboss.threads.JBossThread.run(JBossThread.java:320)
Caused by: javax.naming.NamingException: WFLYNAM0020: Could not resolve service service jboss.naming.context.java.modelSessionFactory
	at org.jboss.as.naming.WritableServiceBasedNamingStore.unbind(WritableServiceBasedNamingStore.java:136)
	at org.jboss.as.naming.NamingContext.unbind(NamingContext.java:318)
	at org.jboss.as.naming.InitialContext$DefaultInitialContext.unbind(InitialContext.java:280)
	at javax.naming.InitialContext.unbind(InitialContext.java:445)
	at javax.naming.InitialContext.unbind(InitialContext.java:445)
	at org.hibernate.engine.jndi.internal.JndiServiceImpl.unbind(JndiServiceImpl.java:210)
	... 11 more

 [0m [0m19:08:24,958 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 60) WFLYCLINF0003: Stopped client-mappings cache from ejb container


```

--- a/feature-pack/src/main/resources/modules/system/layers/base/org/apache/openjpa/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/base/org/apache/openjpa/main/module.xml
@@ -24,6 +24,10 @@
 
 <!-- Represents the OpenJPA module  -->
 <module xmlns="urn:jboss:module:1.3" name="org.apache.openjpa">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
     <resources>
         <artifact name="${org.wildfly:jipijapa-openjpa}"/>
     </resources>

--- a/feature-pack/src/main/resources/modules/system/layers/base/org/eclipse/persistence/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/base/org/eclipse/persistence/main/module.xml
@@ -24,6 +24,10 @@
 
 <!-- Represents the EclipseLink module  -->
 <module xmlns="urn:jboss:module:1.3" name="org.eclipse.persistence">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
     <resources>
         <artifact name="${org.wildfly:jipijapa-eclipselink}"/>
     </resources>

--- a/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/4.3/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/4.3/module.xml
@@ -22,6 +22,31 @@
   ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
   -->
 
-<!-- Represents the Hibernate 4.3.x (org.hibernate:main) module  -->
-<module-alias xmlns="urn:jboss:module:1.3" name="org.hibernate" slot="4.3"  target-name="org.hibernate"/>
+<!-- Represents the Hibernate 4.3.x module  -->
+<module xmlns="urn:jboss:module:1.3" name="org.hibernate" slot="4.3">
+   <resources>
+        <artifact name="${org.wildfly:jipijapa-hibernate4-3}" services="import"/>
+    </resources>
 
+    <dependencies>
+        <module name="asm.asm"/>
+        <module name="com.fasterxml.classmate"/>
+        <module name="javax.api"/>
+        <module name="javax.annotation.api"/>
+        <module name="javax.enterprise.api"/>
+        <module name="javax.persistence.api"/>
+        <module name="javax.transaction.api"/>
+        <module name="javax.validation.api"/>
+        <module name="javax.xml.bind.api"/>
+        <module name="org.antlr"/>
+        <module name="org.apache.commons.collections"/>
+        <module name="org.dom4j"/>
+        <module name="org.infinispan" services="import"/>
+        <module name="org.javassist"/>
+        <module name="org.jboss.as.jpa.spi"/>
+        <module name="org.jboss.jandex"/>
+        <module name="org.jboss.logging"/>
+        <module name="org.jboss.vfs"/>
+        <module name="org.hibernate.commons-annotations"/>
+    </dependencies>
+</module>

--- a/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/jipijapa-hibernate4-3/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/jipijapa-hibernate4-3/main/module.xml
@@ -23,12 +23,16 @@
   -->
 
 <module xmlns="urn:jboss:module:1.3" name="org.hibernate.jipijapa-hibernate4-3">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
     <resources>
         <artifact name="${org.wildfly:jipijapa-hibernate4-3}"/>
     </resources>
 
     <dependencies>
-        <module name="org.hibernate"/>
+        <module name="org.hibernate" slot="4.3" />
         <module name="asm.asm"/>
         <module name="com.fasterxml.classmate"/>
         <module name="javax.api"/>

--- a/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/jipijapa-hibernate5/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/jipijapa-hibernate5/main/module.xml
@@ -23,6 +23,10 @@
   -->
 
 <module xmlns="urn:jboss:module:1.3" name="org.hibernate.jipijapa-hibernate5">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
     <resources>
         <artifact name="${org.wildfly:jipijapa-hibernate5}"/>
     </resources>

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/jarfile/persistence.xml
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/jarfile/persistence.xml
@@ -30,7 +30,8 @@
 			<property name="hibernate.hbm2ddl.auto" value="create-drop" />
 			<property name="hibernate.show_sql" value="true" />
 			<property name="hibernate.session_factory_name" value="modelSessionFactory" />
-                        <property name="jboss.as.jpa.providerModule" value="org.hibernate:4.3" />
+          <!-- ensure that the alias for the org:hibernate module can be used -->
+                        <property name="jboss.as.jpa.providerModule" value="org.hibernate:5.0" />
 		</properties>
 	</persistence-unit>
 </persistence>

